docs/source/howto/data.rst

```diff
@@ -260,6 +260,329 @@ Finding data
+
+
+Delete a group
+^^^^^^^^^^^^^^
+From the command line interface:
+
+.. code-block:: console
+
+      $ verdi group delete another_group
+      Are you sure to delete Group<another_group>? [y/N]: y
+      Success: Group<another_group> deleted.
+
+.. important::
+    Any deletion operation related to groups won't affect the nodes themselves.
+    For example if you delete a group, the nodes that belonged to the group will remain in the database.
+    The same happens if you delete nodes from the group -- they will remain in the database but won't belong to the group anymore.
+
+Copy one group into another
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This operation will copy the nodes of the source group into the destination group.
+If the destination group does not yet exist, it will be created automatically.
+
+From the command line interface:
+
+.. code-block:: console
+
+    $ verdi group copy source_group dest_group
+    Success: Nodes copied from group<source_group> to group<dest_group>
+
+From the Python interface:
+
+.. code-block:: ipython
+
+    In [1]: src_group = Group.objects.get(label='source_group')
+
+    In [2]: dest_group = Group(label='destination_group').store()
+
+    In [3]: dest_group.add_nodes(list(src_group.nodes))
+
+
+Examples for using groups
+-------------------------
+
+In this section, we will provide some practical examples of how one can use Groups to structure and organize the nodes in the database.
+
+Example I: grouping structures with a similar property
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Suppose you have a database of structures for which you have performed a set of simulations.
+Your goal is to continue the study only for the structures for which the bandgap is higher than ``1.0 eV``
+To achieve that goal you would first need to make a query that finds such structures and puts them into a group.
+Let's go through the whole process step by step.
+
+1. We start with creating the group ``promising_structures`` in the ``bash`` shell:
+
+.. code-block:: console
+
+    $ verdi group create promising_structures
+    Success: Group created with PK = 8 and name 'promising_structures'
+
+2. Then we do a query that finds the structures which have the property ``bandgap`` higher than  ``1.0``.
+
+.. code-block:: python
+
+    qb = QueryBuilder()
+    qb.append(StructureData,  tag='structure', project='*') # Here we are projecting the entire structure object
+    qb.append(CalcJobNode, with_incoming='structure', tag='calculation')
+    qb.append(Dict, with_incoming='calculation', filters={'attributes.bandgap':{'>':1.0}})
+
+3. Finally, we put all the structures into the ``promising_structures`` group.
+
+.. code-block:: python
+
+    group = load_group(label='promising_structures')
+    group.add_nodes(q.all(flat=True))
+
+.. note::
+
+    It is completely fine to add nodes to the group even if they were already there.
+    Hence the steps 2. and 3. can be repeated as is.
```

Comment:
Any node can be included in a group only once and if it is added again, it is simply ignored.
This means that `add_nodes` can be safely called multiple times, and only nodes that weren't already part of the group, will be added.